src/main/java/core/basesyntax/service/PurchaseHandler.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseHandler implements OperationHandler {
+    private final Map<String, Integer> fruitStore;
+
+    public PurchaseHandler(Map<String, Integer> fruitStore) {
+        this.fruitStore = fruitStore;
+    }
```

Comment:
PurchaseHandler should not contain any info about fruits. It just handles the purchase :) 
The same for the other handlers.